File: python/src/deployment/service.py
from abc import ABC, abstractmethod
import pulumi


class Service(ABC):

    def __init__(self, config: pulumi.Config):
        self.config = config
        self.environment_name = config.require("environmentName")
        self.service_name = config.require("serviceName")

    @abstractmethod
    def deploy(self):
        pass

File: typescript/src/deployment/service.ts
import * as pulumi from "@pulumi/pulumi";

export abstract class Service {
  protected pulumiConfig: pulumi.Config;

  constructor(config: pulumi.Config) {
    this.pulumiConfig = config;
  }

  abstract deploy(): Promise<{ [key: string]: pulumi.Input<string> }>;
}
